cpp/src/parquet/file_reader.h

```diff
@@ -98,6 +99,7 @@ class PARQUET_EXPORT ParquetFileReader {
     virtual void Close() = 0;
     virtual std::shared_ptr<RowGroupReader> GetRowGroup(int i) = 0;
     virtual std::shared_ptr<FileMetaData> metadata() const = 0;
+    virtual std::shared_ptr<PageIndexReader> page_index_reader() = 0;
```

Comment:
nit: should probably be const OR be GetPageIndexReader to be the same as GetRowGroup.   We should document behavior if Page indices are not written.

Reply:
Same comment applies below.

Reply:
Renamed it to GetPageIndexReader and added comment in detail.